eclipse-solargraph-plugin/src/main/java/io/github/pyvesb/eclipse_solargraph/launch/RubyRunTab.java

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+public class RubyRunTab extends AbstractLaunchConfigurationTab {
+
+	private Text programPathText;
+	private Text argumentsText;
+	private Text workingDirectoryText;
+
+	@Override
+	public void createControl(Composite parent) {
+		Composite resComposite = new Composite(parent, SWT.NONE);
+		resComposite.setLayout(new GridLayout(2, false));
+		new Label(resComposite, SWT.NONE).setText("Ruby Script");
+		this.programPathText = new Text(resComposite, SWT.BORDER);
+		this.programPathText.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, true, false));
+		ControlDecoration decoration = new ControlDecoration(programPathText, SWT.TOP | SWT.LEFT);
+		FieldDecoration fieldDecoration = FieldDecorationRegistry.getDefault().getFieldDecoration(
+		        FieldDecorationRegistry.DEC_ERROR);
+		decoration.setImage(fieldDecoration.getImage());
+		this.programPathText.addModifyListener(event -> {
+			setDirty(true);
+			File file = new File(programPathText.getText());
+			if (!file.isFile()) {
+				String errorMessage = "Unknown file";
+				setErrorMessage(errorMessage);
+				decoration.setDescriptionText(errorMessage);
+				decoration.show();
+			} if (!file.canRead()) {
+				String errorMessage = "Non readable file";
+				setErrorMessage(errorMessage);
+				decoration.setDescriptionText(errorMessage);
+				decoration.show();
+			} else {
+				setErrorMessage(null);
+				decoration.hide();
+			}
+			updateLaunchConfigurationDialog();
+		});
+		new Label(resComposite, SWT.NONE).setText("Arguments");
+		this.argumentsText = new Text(resComposite, SWT.BORDER);
+		this.argumentsText.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, true, false));
+		this.argumentsText.addModifyListener(e -> {
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		});
+		new Label(resComposite, SWT.NONE).setText("Working directory");
+		this.workingDirectoryText = new Text(resComposite, SWT.BORDER);
+		this.workingDirectoryText.setLayoutData(new GridData(SWT.FILL, SWT.DEFAULT, true, false));
+		this.workingDirectoryText.addModifyListener(e -> {
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		});
+		setControl(resComposite);
+	}
+
+	@Override
+	public void setDefaults(ILaunchConfigurationWorkingCopy configuration) {
+		// Nothing to do
+	}
+
+	@Override
+	public void initializeFrom(ILaunchConfiguration configuration) {
+		try {
+			String defaultSelectedFile = "";
+			this.programPathText.setText(configuration.getAttribute(RubyRunDelegate.PROGRAM, defaultSelectedFile)); //$NON-NLS-1$
+			this.argumentsText.setText(configuration.getAttribute(RubyRunDelegate.ARGUMENTS, "")); //$NON-NLS-1$
+			this.workingDirectoryText.setText(configuration.getAttribute(DebugPlugin.ATTR_WORKING_DIRECTORY, "")); //$NON-NLS-1$
+		} catch (CoreException e) {
+			e.printStackTrace();
```

Comment:
Same here: should we log about this error?
```suggestion
			LogHelper.error("Exception whilst initializing Ruby tab", e);
```